### ISSUE TITLE
Update issues : pyrevit and extensions update fail and windows is closing even in debug mode

### ISSUE THREAD

**Issue by CyrilWaechter** [eirannejad/pyRevit#334]
Hi,
I get many issues on many machines when updating with standard «Update» tool. It often end by installing git on client computer and use git pull.

I identified some possible issues : 

- when a file is modified : update tool probably don't force update wait a merge ?
- when extension is installed in default folder (%AppData%\pyRevit\pyRevit-v4\pyRevit\extension\[extension name]. Maybe it interferes with git tracking (a repo inside another repo might it generate some issues ?).

I often get the following message but windows automatically close even when in debug mode so It's difficult to investigate as proposed.

> [extension name] update failed. Run in debug mode to see why

**Comment by eirannejad** [comment]
**Ends by installing git client????**

The standard update tool doesn't force hard reset before update so if any changes are made to pyRevit core and tools, they should remain unchanged unless there are merge conflicts. Any extensions other than default inside the default extension folder have been ignored in `.gitignore` as well.

I'm working on a `pyrevit` CLI tool that handles all repo and configuration tasks around pyRevit. Below is a snapshot of functionality. This tool will be used in all pyRevit updates and also allows for easier remote management of pyRevit deployments.

![2018-05-07 10_36_57-pyrevitlabs - microsoft visual studio](https://user-images.githubusercontent.com/8197916/39715593-abf97b98-51e2-11e8-962c-2f908b8c4a65.png)

**Comment by CyrilWaechter** [comment]
Yes git client for windows : https://git-scm.com/downloads
I install it, launch Git bash inside pyRevit then perform a git pull (and git clean or git revert if some files have been changed).

Thanks for clarifications about .extension folders in `.gitignore`.

I know that 1 or 2 times update failed because I modified `extensions.json` or a script to fast correct a bug but some other times I just didn't get what could have blocked the update process. I'll wait your CLI tools to see if it helps and if there is some remaining cases of failure.

But it would still be nice that the report window do not close if any update fail or if run in debug mode.
